### ISSUE TITLE
Skirmish Mechbay fails to load due to NRE

### ISSUE THREAD

**Issue by IceRaptor** [BattletechModders/IttyBittyLivingSpace#1]
From ticket-0575:

HarkonnenLast Monday at 3:36 PM
@CptMoore  you fixed this once, but it returned. StructureFactor items again shows zero weight in tooltip in mechbay.

```
2019-05-29T07:11:06 FYLS [ERROR] System.NullReferenceException: Object reference not set to an instance of an object

...and core weight is noy adjusted by engine either

...hmm... did you check on full RT or just ME alone? there are some IttyBittyLivingSpace NREs in skirmish mechbay, maybe its interfering somehow?

at IttyBittyLivingSpace.Helper.GetGearInventorySize (BattleTech.SimGameState) <0x00036>
at IttyBittyLivingSpace.TooltipPrefab_Equipment_SetData.Postfix (BattleTech.UI.Tooltips.TooltipPrefab_Equipment,object,TMPro.TextMeshProUGUI) <0x000d5>
at (wrapper dynamic-method) BattleTech.UI.Tooltips.TooltipPrefab_Equipment.SetData_Patch2 (object,object) <0x00790>
at BattleTech.UI.Tooltips.TooltipManager.SetActiveTooltip (object,string) <0x000c6>
at BattleTech.UI.Tooltips.TooltipManager.SpawnTooltip (object,ulong,ulong,single,string,bool) <0x0013f>
at BattleTech.UI.Tooltips.HBSTooltip.OnPointerEnter (UnityEngine.EventSystems.PointerEventData) <0x00258>
at UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IPointerEnterHandler,UnityEngine.EventSystems.BaseEvent
```

**Comment by IceRaptor** [comment]
[cleaned_output_log (2).txt](https://github.com/IceRaptor/IttyBittyLivingSpace/files/3233180/cleaned_output_log.2.txt)

**Comment by IceRaptor** [comment]
Fixed in 0.3.7

**Comment by CptMoore** [comment]
@IceRaptor I dont think you fix the NPE, because ii crashes in GetGearInventorySize and there are many more possible NPE in there.

In any case, I suggest adding try { } catch () around all your patches so you don't kill off other mods patches!

What I do:
```
public static void Postfix(MechDef mechDef, ref float currentValue, ref float maxValue)
{
	try
	{
		currentValue += WeightsHandler.Shared.TonnageChanges(mechDef);
		currentValue = Mathf.Min(currentValue, maxValue);
	}
	catch (Exception e)
	{
		Control.mod.Logger.LogError(e);
	}
}
```

**Comment by IceRaptor** [comment]
@CptMoore just to be clear - are you saying you still have an issue after the fix in 0.3.7, or you think you still will because the fix was insufficient?

All I did in 0.3.7 was check that simGame was active, which shouldn't be present if Skirmish is around. That should bypass all of the IBLS logic because IBLS only makes sense in a SimGameSetting. 

So yes, absolutely - there are many possible NREs in GetGearInventory size. They should only impact SimGame, but yes, I'll look into correcting them and possibly catching the exception. Given how non-critical these patches are, that's a safe approach. 

I just want to make sure you're no longer seeing issues in *skirmish* in particular, because I'm disabling the patch during that flow.

Cheers, and thank you for the report!

**Comment by CptMoore** [comment]
> you think you still will because the fix was insufficient?

just that, tbh I dont play RT (too long loading times) so I dont know if it will occur again. lets see if it will be reported once more after the UW mess
